### PR TITLE
fix: load semester during calendar parsing

### PR DESCRIPTION
**Pull request by dfarooq610**
# Description
`course.semester` was nonexistent when parsing the ical, so we loaded the semester in. Also, fixed up the seeds controller for the SemesterModel so we dont have infinite semesters locally. 

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?
Imported my own Ical File (not in this PR bc it's a secret), and manually hit the endpoint to update the calendar. We replicated the error, and after making these changes, we could successfully parse the calendar.

### CODE REVIEW

**Review by ashdawngary — APPROVED** [review]
looks good

File: packages/server/src/course/ical.service.ts [hunk 1]
```diff
@@ -190,6 +190,10 @@ export class IcalService {
     let queue = await QueueModel.findOne({
       where: { courseId: course.id, room: 'Online' },
     });
+    const semester =
+      (await SemesterModel.findOne({
+        where: { id: course.semesterId },
+      })) || course.semester;
```
Comment (IrisLiu-00) [inline packages/server/src/course/ical.service.ts:196]: is `course.semester` supposed to go first in the `OR` here? this way we dont run the db query if we already have the sem.

**Review by IrisLiu-00 — APPROVED** [review]
Thanks for getting this, guys!! 🙏 🙏

File: packages/server/src/course/ical.service.ts (hunk 1 above)
Comment (ashdawngary) [reply packages/server/src/course/ical.service.ts:196]: yes i can fix that -- in prod course.semester is null its just the factories that dont give a semesterid but give a semestermodel object

**Pull request merged**